sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -213,6 +215,34 @@ case class Literal protected (value: Any, dataType: DataType)
       }
     }
   }
+
+  override def sql: Option[String] = Option((value, dataType) match {
+    case _ if value == null =>
+      "NULL"
```

Comment:
Will we lose data type for this null literal? Should we create a cast if the data type is not `NullType`?